AG-802: Close hamburger menu when menu item selected

JessterB opened the pull request:
https://sagebionetworks.jira.com/browse/AG-802

JessterB committed:
AG-802: Close hamburger menu when menu item selected

jay-hodgson reviewed (approved):
Out of curiosity, would this also work if you set a single click handler on the ul element instead?

JessterB commented:
Yes, yes it does. Good call, I've updated the PR.

JessterB commented:
After testing that change a bit more, I realized that it prevents you from using the search box in the hamburger menu, since the menu closes when you click on it. I've reverted the change and we will stick with the two separate click handlers.
I also discovered that the menu would close if you clicked on whitespace rather than on a navigation link, so I've moved the new click handler down to the anchor level to prevent that.

JessterB committed:
Move click handler to anchor to prevent closing menu when whitespace is clicked
